src/Features/CSharp/Portable/GenerateMember/GenerateVariable/CSharpGenerateVariableService.cs

```diff
@@ -156,11 +157,24 @@ protected override bool TryConvertToLocalDeclaration(ITypeSymbol type, SyntaxTok
             return false;
         }
 
-        private static TypeSyntax GenerateTypeSyntax(ITypeSymbol type, OptionSet options)
+        private static TypeSyntax GenerateTypeSyntax(ITypeSymbol type, OptionSet options, ExpressionSyntax expression, SemanticModel semanticModel, CancellationToken cancellationToken)
         {
-            return type.ContainsAnonymousType() || (options.GetOption(CSharpCodeStyleOptions.UseVarWhenDeclaringLocals) && type.TypeKind != TypeKind.Delegate)
-                ? SyntaxFactory.IdentifierName("var")
-                : type.GenerateTypeSyntax();
+            // if there isn't a semantic model, we cannot perform further analysis.
+            if (semanticModel != null)
```

Comment:
This looks like it will try to GenerateTypeSyntax if the type does contain anonymous types and semanticModel is null.  Is that a problem in practice?  Can you just require SemanticModel?